service/history/workflow/update/registry.go

```diff
@@ -219,6 +227,51 @@ func (r *registry) FindOrCreate(ctx context.Context, id string) (*Update, bool,
 	return upd, false, nil
 }
 
+func (r *registry) TryResurrect(ctx context.Context, acptOrRejMsg *protocolpb.Message) (*Update, error) {
+	if acptOrRejMsg == nil || acptOrRejMsg.Body == nil {
+		return nil, nil
+	}
+
+	// Check only total limit here. This might add more than maxInFlight updates to registry,
+	// but provides better developer experience.
+	if err := r.checkTotalLimit(ctx); err != nil {
+		return nil, err
+	}
+
+	body, err := acptOrRejMsg.Body.UnmarshalNew()
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
+	}
+	err = utf8validator.Validate(body, utf8validator.SourceRPCRequest)
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
+	}
+	var reqMsg *updatepb.Request
+	switch updMsg := body.(type) {
+	case *updatepb.Acceptance:
+		reqMsg = updMsg.GetAcceptedRequest()
+	case *updatepb.Rejection:
+		reqMsg = updMsg.GetRejectedRequest()
+	}
+	if reqMsg == nil {
+		return nil, nil
+	}
+	reqAny, err := anypb.New(reqMsg)
+	if err != nil {
+		return nil, invalidArgf("unable to marshal request: %v", err)
```

Comment:
unmarshal?

whatg causes this error?

Reply:
"unmarshal" yes (again, copied from other place, fixed it there too). The cause would be very weird, like manually built proto message w/o setting type.